### PR TITLE
!rem #22073 enable mutual TLS authentication by default

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: akka-remote/src/main/resources/reference.conf [hunk 1]
```diff
@@ -641,17 +641,17 @@ akka {
...
-        # However, starting from the version this setting was added, even with this setting "off", the active side
-        # (TLS client side) will use the given key-store to send over a certificate if asked. A rolling upgrades from
-        # older versions of Akka can therefore work like this:
-        #   - upgrade all nodes to an Akka version supporting this flag, keeping it off
-        #   - then switch the flag on and do again a rolling upgrade of all nodes
+        # However, starting with Akka 2.4.12, even with this setting "off", the active side (TLS client side)
+        # will use the given key-store to send over a certificate if asked. A rolling upgrade from versions of
+        # Akka < 2.4.12 can therefore work like this:
+        #   - upgrade all nodes to an Akka version >= 2.4.12, in the best case the latest version, but keep this setting at "off"
+        #   - then switch this flag to "on" and do again a rolling upgrade of all nodes
```
Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf]: That's great. I was thinking if we had to mention something in akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst, but perhaps it's not even needed or what do you think?

Comment (jrudolph) [inline akka-remote/src/main/resources/reference.conf]: Yes, I guess makes sense if people update from an older Akka version. Will add that.

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 2]
```diff
@@ -69,34 +60,59 @@ which explains using and implementing GraphStages in more practical terms than t
...
-As users were often confused by "when to use an Actor vs. when to use an Agent?" a decision was made to deprecate
-the Agents, as they rarely are really enough and do not fit the Akka spirit of thinking about distribution.
-We also anticipate to replace the uses of Agents by the upcoming Akka Typed, so in preparation thereof the Agents have been deprecated in 2.5.
+Nodes that are configured with this setting to 'on' might not be able to receive messages from nodes that run on older
+versions of akka-remote. This is because in versions of Akka < 2.4.12 the active side of the remoting
+connection will not send over certificates even if asked to.
 
-If you use Agents and would like to take over the maintanance thereof, please contact the team on gitter or github.
+It is still possible to make a rolling upgrade from a version < 2.4.12 by doing the upgrade stepwise:
+ * first, upgrade Akka to the latest version but keep `akka.remote.netty.ssl.require-mutual-authentication` at `off`
```
Comment (ktoso) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: double `` around the snippet, we're still on restructured text here

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 3]
```diff
@@ -69,34 +60,59 @@ which explains using and implementing GraphStages in more practical terms than t
...
+   and do a first rolling upgrade
+ * second, turn the setting to "on" and do another rolling upgrade
 
+For more information see the documentation for the `akka.remote.netty.ssl.require-mutual-authentication` configuration setting
```
Comment (ktoso) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: here too

**Review by ktoso — APPROVED** [review]
LGTM, nothing more to add here I think.

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst (hunk 2 above)
Comment (jrudolph) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: Ooh, good catch.

**Comment by jrudolph** [comment]
Fixed the quotes.

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 4]
```diff
@@ -4,23 +4,14 @@
 Migration Guide 2.4.x to 2.5.x
 ##############################
 
-Akka Persistence
-================
-
-Removal of PersistentView
--------------------------
+Actor Actor
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: Actor Actor :)

**Review by patriknw — APPROVED** [review]
LGTM, fix the typo and then merge

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst (hunk 4 above)
Comment (jrudolph) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: Oops, stupid.

**Comment by jrudolph** [comment]
Refs #22073.

**Pull request merged**